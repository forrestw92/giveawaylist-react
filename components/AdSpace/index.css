.advspace {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.advcard {
  display: -ms-flexbox;
  display: flex;
  flex-grow: 1;
  flex-flow: column nowrap;
  list-style: none;
  padding: 0;
  height: 200px;
  max-height: 479px;
  margin: 20px;
  position: relative;
  background: #fff;
  overflow: hidden;
  box-shadow: 0 0 6px 0px #f28a2c;
  transition: all 0.3s ease;
  width: 100%;
  min-width: 250px;
  max-width: 300px;
}
.advtext {
  display: flex;
  text-decoration: underline;
  text-decoration-color: #f28a2c;
  justify-content: center;
  text-align: center;
  margin: auto;
  font-size: 2em;
}
