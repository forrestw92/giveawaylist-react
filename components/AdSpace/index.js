import React from "react";
import Link from "next/link";

import stylesheet from "./index.css";
function AdSpace() {
  return (
    <div className="advspace">
      <div className="advcard">
        <Link href={"/contact"} shallow>
          <a href={"/contact"} className="advtext">
            Advertise Here!
          </a>
        </Link>
      </div>
      <style jsx>{stylesheet}</style>
    </div>
  );
}
export default AdSpace;
