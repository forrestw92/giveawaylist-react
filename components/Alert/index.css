.alert {
  position: fixed;
  top: 75px;
  right: 10px;
  padding: 5px 15px;
  box-shadow: 0 0 5px 1px rgba(0, 0, 0, 0.25);
  z-index: 11;
  opacity: 1;
  cursor: default;
}
.primary {
  background: #1d3557;
  color: #fdfefd;
}
.danger {
  color: #111111;
  background: rgb(230, 57, 70);
}
.info {
  color: #111111;
  background: #a8dadc;
}
.hide {
  opacity: 0;
}
