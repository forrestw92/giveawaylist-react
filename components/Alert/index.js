import React from "react";
import { string, node, bool, func, number } from "prop-types";
import stylesheet from "./index.css";
class Alert extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      forceHide: false
    };
  }

  componentDidMount() {
    const { ttl, onDeath } = this.props;
    setTimeout(() => {
      this.setState({ forceHide: true }, () => onDeath());
    }, ttl * 1000);
  }

  render() {
    const { alertType, children, show } = this.props;
    return (
      <div
        className={`alert ${alertType ? alertType : "primary"} ${
          show && !this.state.forceHide ? "" : "hide"
        }`}
        role="alert"
        aria-atomic="true"
      >
        {children}
        <style jsx>{stylesheet}</style>
      </div>
    );
  }
}

Alert.propTypes = {
  ttl: number,
  alertType: string,
  show: bool.isRequired,
  onDeath: func.isRequired,
  children: node.isRequired
};

export default Alert;
