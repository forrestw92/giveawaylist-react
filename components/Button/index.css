.button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  height: 40px;
  width: 125px;
  text-align: center;
  dispaly: flex;
  justify-content: center;
  font-size: 18px;
  box-shadow: 0 1px 1px 0 rgb(0, 0, 0);
  transition: box-shadow 0.25s ease;
  align-items: center;
}
.button.login,
.button.register {
  display: flex;
  margin: auto;
  color: white;
  text-decoration: none;
  background: #1d3557;
}
.button.login:hover,
.button.register:hover {
  box-shadow: 0 3px 1px 1px rgba(0, 0, 0, 0.8);
}
