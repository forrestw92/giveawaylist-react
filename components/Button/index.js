import React from "react";
import { string, func } from "prop-types";
import stylesheet from "./index.css";
function Button(props) {
  const { className, label, _onClick, type, href } = props;
  return (
    <React.Fragment>
      {type === "button" ? (
        <button className={`button ${className}`} onClick={e => _onClick(e)}>
          {label}
        </button>
      ) : (
        <a href={href} className={`button ${className}`}>
          {label}
        </a>
      )}
      <style jsx>{stylesheet}</style>
    </React.Fragment>
  );
}

Button.propTypes = {
  _onClick: func.isRequired,
  label: string.isRequired,
  className: string.isRequired,
  type: string.isRequired,
  href: string
};

export default Button;
