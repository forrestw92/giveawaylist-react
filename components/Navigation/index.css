nav {
  display: flex;
  flex-flow: row nowrap;
  max-width: 960px;
  width: 100%;
  justify-content: space-around;
}
.link {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
}
.link--image {
  margin-right: 10px;
}
.logo {
  display: flex;
  height: 50px;
}

@media only screen and (max-width: 425px) {
  .logo {
    left: 5px;
  }
}

.navigation {
  margin: 0;
  padding: 0;
  display: flex;
  flex-flow: row nowrap;
  height: 50px;
}
.navigation .nav--item {
  margin: 15px;
  display: flex;
  align-items: center;
  font-size: 20px;
  background: #1d3557;
  justify-content: space-evenly;
  position: relative;
}

.navigation .nav--item:hover {
  color: #fdfefd;
}
.navigation .nav--item:hover a {
  color: #a8dadc;
}

.navigation .nav--item a {
  color: #fdfefd;
  text-align: center;
  margin: auto;
}
.navigation .nav--item .active {
  color: #f28a2c;
}
@media only screen and (min-width: 676px) and (max-width: 767px) {
  .link--image {
    margin: 0;
  }
  .link {
    flex-flow: column wrap;
  }
  .navigation .nav--item a {
    margin: 0;
  }
}
@media only screen and (max-width: 675px) {
  .navigation {
    background: #1d3557;
    width: 100vw;
    z-index: 10;
    position: fixed;
    bottom: 0;
    justify-content: center;
  }
}
@media only screen and (max-width: 548px) {
  .link--image {
    margin: 0;
  }
  .link {
    flex-flow: column wrap;
  }
  .navigation .nav--item a {
    margin: 0;
  }
}
@media only screen and (max-width: 330px) {
  .navigation .nav--item {
    font-size: 16px;
  }
}
