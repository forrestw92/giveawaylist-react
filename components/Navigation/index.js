import React from "react";
import Link from "next/link";
import { connect } from "react-redux";
import { bool, string } from "prop-types";
import stylesheet from "./index.css";
class Navigation extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      links: [
        {
          href: "/",
          label: "Home",
          shouldRender: true,
          className: "nav--item",
          image: "../../static/icons/home.svg"
        },
        {
          href: "/ending",
          label: "Ending",
          shouldRender: true,
          className: "nav--item",
          image: "../../static/icons/clock.svg"
        },
        {
          href: "/saved",
          label: "Saved",
          shouldRender: false,
          className: "nav--item",
          image: "../../static/icons/saved.svg"
        },
        {
          href: "/ebooks",
          label: "eBooks",
          shouldRender: true,
          className: "nav--item",
          image: "../../static/icons/book.svg"
        },
        {
          href: "/sweepstakes",
          label: "Sweepstakes",
          shouldRender: false,
          className: "nav--item"
        },
        {
          href: "/profile",
          label: "Profile",
          shouldRender: true,
          className: "nav--item",
          image: "../../static/icons/user.svg"
        },
        {
          href: "/profile/login",
          label: "Login",
          shouldRender: false,
          className: "nav--item",
          image: "../../static/icons/log-in.svg"
        },
        {
          href: "/profile/register",
          label: "Register",
          shouldRender: false,
          className: "nav--item",
          image: "../../static/icons/user-plus.svg"
        }
      ]
    };
  }
  static getDerivedStateFromProps(prevProps, state) {
    if (prevProps.currentPage === "/profile/login") {
      return state.links.map(link => {
        if (link.href === "/profile/login") {
          link.shouldRender = true;
        }
        if (link.href === "/profile") {
          link.shouldRender = false;
        }
        if (link.href === "/profile/register") {
          link.shouldRender = false;
        }
      });
    }
    if (prevProps.currentPage === "/profile/register") {
      return state.links.map(link => {
        if (link.href === "/profile/register") {
          link.shouldRender = true;
        }
        if (link.href === "/profile") {
          link.shouldRender = false;
        }
        if (link.href === "/profile/login") {
          link.shouldRender = false;
        }
      });
    }
    return null;
  }

  componentDidMount() {
    if (this.props.loggedIn) {
      const links = this.state.links.map(item => {
        if (item.href === "/saved") {
          item.shouldRender = !item.shouldRender;
        }
        return item;
      });
      this.setState({ links });
    }
  }

  render() {
    const renderLinks = this.state.links.filter(link => link.shouldRender);
    const { currentPage } = this.props;
    return (
      <nav role="navigation">
        <img
          src="../../static/logo.svg"
          className={stylesheet["logo"]}
          alt="Giveaway List Logo"
        />
        <ul className={stylesheet["navigation"]} id={"menu"} tabIndex={"-1"}>
          {renderLinks.map(({ href, label, image, className }) => (
            <li key={href} className={`${stylesheet[className]}`}>
              <Link prefetch href={href}>
                <a
                  className={
                    currentPage === href
                      ? `${stylesheet["link"]} ${stylesheet["active"]}`
                      : stylesheet["link"]
                  }
                >
                  <img src={image} className={stylesheet["link--image"]} />
                  <span>{label}</span>
                </a>
              </Link>
            </li>
          ))}
        </ul>
      </nav>
    );
  }
}
Navigation.propTypes = {
  loggedIn: bool.isRequired,
  currentPage: string.isRequired
};
function mapStateToProps(state) {
  return {
    loggedIn: state.user.loggedIn,
    currentPage: state.nav.currentPage
  };
}
export default connect(mapStateToProps)(Navigation);
