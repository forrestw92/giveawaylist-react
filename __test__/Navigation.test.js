/* eslint-env jest */

import { render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Router from "next/router";
import { Navigation } from "../components/Navigation";
const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe("With Enzyme", () => {
  it("Navigation Has Logo", () => {
    jest.mock("next/router");
    const app = render(<Navigation currentPage={"/"} loggedIn={false} />);
    expect(app.find("img").attr("src")).toEqual("../../static/logo.svg");
  });
});

describe("With Snapshot Testing", () => {
  it("Navigation Has Logo", () => {
    const component = renderer.create(
      <Navigation currentPage={"/"} loggedIn={false} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
