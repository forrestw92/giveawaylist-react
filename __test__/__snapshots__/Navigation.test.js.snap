// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Navigation Has Logo 1`] = `
<nav
  role="navigation"
>
  <img
    alt="Giveaway List Logo"
    className="logo"
    src="../../static/logo.svg"
  />
  <ul
    className="navigation"
    id="menu"
    tabIndex="-1"
  >
    <li
      className="nav--item"
    >
      <a
        className="link active"
        href="/"
        onClick={[Function]}
      >
        <img
          className="link--image"
          src="../../static/icons/home.svg"
        />
        <span>
          Home
        </span>
      </a>
    </li>
    <li
      className="nav--item"
    >
      <a
        className="link"
        href="/ending"
        onClick={[Function]}
      >
        <img
          className="link--image"
          src="../../static/icons/clock.svg"
        />
        <span>
          Ending
        </span>
      </a>
    </li>
    <li
      className="nav--item"
    >
      <a
        className="link"
        href="/ebooks"
        onClick={[Function]}
      >
        <img
          className="link--image"
          src="../../static/icons/book.svg"
        />
        <span>
          eBooks
        </span>
      </a>
    </li>
    <li
      className="nav--item"
    >
      <a
        className="link"
        href="/profile"
        onClick={[Function]}
      >
        <img
          className="link--image"
          src="../../static/icons/user.svg"
        />
        <span>
          Profile
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
